lib/pbench/test/unit/server/test_datasets_list.py

```diff
@@ -93,12 +103,25 @@ def get_results(self, name_list: List[str]) -> JSON:
                     },
                 }
             )
-        return {"results": list}
+        query["page_no"] = page_no + 1
+        if len(paginated_name_list) < page_limit:
+            next_url = ""
+        else:
+            next_url = (
+                f"http://localhost{server_config.rest_uri}/datasets/list"
+                + "?"
+                + urlencode(query)
+            )
```

Comment:
Given that you're already using an f-string, you might as well include the `?` in line 111, rather than using an extra concatenation.

Reply:
Actually, I think I like [Dave's approach](https://github.com/distributed-system-analysis/pbench/pull/2855#discussion_r882030796) better, although it is sort of hard to immediately comprehend.